### PR TITLE
Fix #629 avoid polygons restrictions

### LINKED ISSUE #629: Restrict polygon properties for avoid_polygons
To prevent our servers from crashing, we should restrict some polygon property/properties a user can pass with `avoid_polygon` parameter. That should also go in the `app.config` of course.

Possible property restriction:
- area
- extent in X or Y, e.g. no more than 10 km in X OR Y

### PR DESCRIPTION
**Pull request by takb**
### Pull Request Checklist

- [x] 1. I have [**rebased**](https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines) the latest version of the development branch into my feature branch and all conflicts have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [x] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [x] 6. I have created API tests for any new functionality exposed to the API.
- [x] 7. If changes/additions are made to the app.config file, I have added these to the app.config wiki page on github along with a short description of what it is for, and documented this in the Pull Request (below).
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset (at least Germany) and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the importer etc.), I have generated longer distance routes for the affected profiles with different options (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end points generated from the current live ORS. If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage and why the change was needed.

Fixes #629 .

### Information about the changes
- Key functionality added:
Parameters and checks to limit area & extent

### Required changes to app.config (if applicable)
- OPTIONAL parameter in ors.services.routing.profiles.default_params:
`maximum_avoid_polygon_area`: area in square kilometers, unlimited if omitted
`maximum_avoid_polygon_extent`: extent in kilometers, unlimited if omitted

The updated test requires the params values as set in the `app.config.test` and `app.config.sample` files.

### CODE REVIEW

**Comment by takb** [comment]
@nilsnolde still has to answer my question from the issue
> what are the values that endanger server stability?

**Comment by nilsnolde** [comment]
Thanks for the bump @takb , somehow I missed the issue conversation.. 

As @HendrikLeuschner pointed out, it's hard to restrict reasonably, but area and extent in either direction are an OK proxy. Maybe we can try 20 km for extent and 200 km2 for area? Or what do you mean with values?

How is `extent` it implemented @takb ? If I put 20000 m, will it not allow polygons which are > 20 km in **either** X or Y?

BTW, we shouldn't forget the Docker app.config.sample when we change defaults or parameters.

File: openrouteservice-api-tests/conf/app.config.test [hunk 1]
```diff
@@ -52,6 +52,8 @@
             maximum_segment_distance_with_dynamic_weights: 50000,
             maximum_waypoints: 50,
             maximum_snapping_radius: 350,
+            maximum_avoid_polygon_area: 2.5,
+            maximum_avoid_polygon_extent: 1.5,
```
Comment (nilsnolde) [inline openrouteservice-api-tests/conf/app.config.test]: What's the units?

**Review by nilsnolde — COMMENTED** [review]
Won't do a code review. Just wondering about the units of the parameters. Mollweide :+1:

File: openrouteservice-api-tests/conf/app.config.test (hunk 1 above)
Comment (takb) [reply openrouteservice-api-tests/conf/app.config.test]: square kilometers / kilometers, and yes, the extent checks both x and y.

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/routing/ParamsTest.java [hunk 2]
```diff
@@ -700,6 +701,53 @@ public void expectAvoidpolygonsTypeError() {
...
+				.header("Accept", "application/json")
+				.header("Content-Type", "application/json")
+				.pathParam("profile", getParameter("carProfile"))
+				.body(body.toString())
+				.when()
+				.post(getEndPointPath() + "/{profile}")
+				.then()
+				.assertThat()
+				.body("error.code", is(2003))
+				.body("error.message", is("The extent of a polygon to avoid must not exceed 1.5 kilometers."))
```
Comment (sfendrich) [inline openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/routing/ParamsTest.java]: Shouldn't the 1.5 km be drawn from the config?

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/routing/ParamsTest.java [hunk 3]
```diff
@@ -700,6 +701,53 @@ public void expectAvoidpolygonsTypeError() {
...
+				.header("Accept", "application/json")
+				.header("Content-Type", "application/json")
+				.pathParam("profile", getParameter("carProfile"))
+				.body(body.toString())
+				.when()
+				.post(getEndPointPath() + "/{profile}")
+				.then().log().all()
+				.assertThat()
+				.body("error.code", is(2003))
+				.body("error.message", is("The area of a polygon to avoid must not exceed 2.5 square kilometers."))
```
Comment (sfendrich) [inline openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/routing/ParamsTest.java]: Shouldn't the 2.5 sqkm be drawn from the config?

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/routing/ParamsTest.java (hunk 2 above)
Comment (takb) [reply openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/routing/ParamsTest.java]: you're right, fixed.

File: openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/routing/ParamsTest.java (hunk 3 above)
Comment (takb) [reply openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/routing/ParamsTest.java]: ditto

**Comment by takb** [comment]
> BTW, we shouldn't forget the Docker app.config.sample when we change defaults or parameters.
@nilsnolde , is that a different file than the app.config.sample in the ORS project resources? if so, where do I find it?

File: openrouteservice-api-tests/conf/app.config.test (hunk 1 above)
Comment (nilsnolde) [reply openrouteservice-api-tests/conf/app.config.test]: I'd rather stay with meters as units for consistency's sake. All limiting distance/area values are in meters.

**Comment by nilsnolde** [comment]
https://github.com/GIScience/openrouteservice/tree/master/docker/conf

I'll have a look this/next week, if we can simplify that and only have one app.config. But I don't think so, at least not without sacrificing a lot of convenience/functionality or adding a lot of confusion.

**Review by sfendrich — APPROVED** [review]

File: openrouteservice-api-tests/conf/app.config.test (hunk 1 above)
Comment (takb) [reply openrouteservice-api-tests/conf/app.config.test]: ok, changed.

**Pull request merged**